eventarc/generic/main.go

```diff
@@ -24,34 +24,32 @@ import (
 	"log"
 	"net/http"
 	"os"
+	"strings"
 )
 
-func logAndRespond(w http.ResponseWriter, msg string) {
-	log.Println(msg)
-	fmt.Fprintln(w, msg)
-}
-
 // GenericHandler receives and echos a HTTP request's headers and body.
 func GenericHandler(w http.ResponseWriter, r *http.Request) {
-	logAndRespond(w, "Event received!")
+	log.Println("Event received!")
 
 	// Log all headers besides authorization header
-	logAndRespond(w, "HEADERS:")
-	delete(r.Header, "Authorization")
+	log.Println("HEADERS:")
 	headerMap := make(map[string]string)
 	for k, v := range r.Header {
-		headerMap[k] = string(v[0])
-		logAndRespond(w, fmt.Sprintf("%q: %q\n", k, v[0]))
+		val := strings.Join(v, ",")
+		if k != "Authorization" {
+			headerMap[k] = string(val)
+			log.Println(fmt.Sprintf("%q: %q\n", k, val))
+		}
```

Comment:
```suggestion
		if k != "Authorization" {
			val := strings.Join(v, ",")
			headerMap[k] = val
			log.Println(fmt.Sprintf("%q: %q\n", k, val))
		}
```

You could also make `headerMap` a `map[string][]string` and not have to do `strings.Join`.